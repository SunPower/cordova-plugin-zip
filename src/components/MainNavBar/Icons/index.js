export { ReactComponent as AlertsIcon } from './alerts.svg'
export { ReactComponent as HamburgerMenu } from './hamburger-menu.svg'
export { ReactComponent as Profile } from './profile.svg'
