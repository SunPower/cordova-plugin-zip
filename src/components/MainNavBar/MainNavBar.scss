@import "../../styles/variables";

.main-nav-bar {
  background-color: $navbar-black !important;
  padding-top: 5px;
  min-height: 3rem !important;

  .header-container {
    margin: 0 !important;
    flex-grow: 1;
    justify-content: space-between;

    > div:first-of-type {
      max-width: 50px;
    }

    .header {
      align-items: center;
      justify-content: center;
      font-size: $font-size !important;
      line-height: 0.75rem !important;

      .navbar-header {
        font-weight: $bold-text;
        letter-spacing: $letter-spacing-big;
        color: $white;
        text-transform: uppercase;
      }
    }

    .actions {
      display: flex;
      flex-direction: row;
      justify-content: space-between;
      padding-right: 22px;
      padding-left: 0;
      min-width: 72px;
      max-width: 80px;

      svg {
        flex-grow: 1;
      }
    }

    .alerts-bell {
      position: relative;
    }

    .has-alerts {
      width: 12px;
      height: 12px;
      top: 0;
      right: 0;
      border-radius: 100%;
      border: 2px solid $white;
      position: absolute;
      background: $orange-default;
    }
  }
}
