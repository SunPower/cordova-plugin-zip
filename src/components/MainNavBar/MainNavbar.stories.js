import React from 'react'

import { storiesOf } from '@storybook/react'

import MainNavbar from './index'

storiesOf('MainNavbar', module).add('Simple', () => (
  <MainNavbar location={{ pathname: '/' }} />
))
