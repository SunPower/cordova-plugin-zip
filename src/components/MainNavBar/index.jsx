import React from 'react'
import { Link } from 'react-router-dom'
import { useSelector } from 'react-redux'
import { AlertsIcon, HamburgerMenu, Profile } from './Icons'
import './MainNavBar.scss'
import paths from '../../pages/Router/paths'

function MainNavBar({ location }) {
  const hasAlerts = useSelector(
    state =>
      state.alerts &&
      Object.keys(state.alerts.data).filter(
        k => !state.alerts.data[k].seenTimestamp
      ).length >= 0
  )
  const userName = useSelector(
    state => state.user && state.user.data && state.user.data.firstName
  )
  const title = userName ? `${userName}'s home` : 'home'

  return (
    <nav
      className="main-nav-bar navbar"
      role="navigation"
      aria-label="main navigation"
    >
      <div className="header-container columns is-mobile">
        <div className="column">
          <Link
            className="link"
            to={{
              pathname: paths.MENU,
              state: {
                from: location.pathname
              }
            }}
          >
            <HamburgerMenu />
          </Link>
        </div>
        <div className="column is-two-thirds header is-flex">
          <h1 className="navbar-header">{title}</h1>
        </div>
        <div className="column actions">
          <Link
            to={{
              pathname: paths.NOTIFICATIONS,
              state: {
                from: location.pathname
              }
            }}
          >
            <div className="alerts-bell pr-5">
              <AlertsIcon />
              <div className={hasAlerts ? 'has-alerts' : 'is-hidden'}></div>
            </div>
          </Link>
          <Link
            to={{
              pathname: paths.PROFILE,
              state: {
                from: location.pathname
              }
            }}
          >
            <div className="pl-5">
              <Profile />
            </div>
          </Link>
        </div>
      </div>
    </nav>
  )
}

export default MainNavBar
