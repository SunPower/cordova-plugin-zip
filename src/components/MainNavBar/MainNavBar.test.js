import React from 'react'
import * as reactRedux from 'react-redux'
import { shallow } from 'enzyme'
import MainNavBar from '.'

describe('MainNavBar component', () => {
  const previousLocation = {
    pathname: '/'
  }

  it('renders without crashing', () => {
    jest.spyOn(reactRedux, 'useSelector').mockImplementation(() => true)
    const component = shallow(<MainNavBar location={previousLocation} />)
    expect(component).toMatchSnapshot()
  })

  it('hides bell notification if there are no alerts', () => {
    jest.spyOn(reactRedux, 'useSelector').mockImplementation(() => false)
    const component = shallow(<MainNavBar location={previousLocation} />)
    expect(component).toMatchSnapshot()
  })
})
