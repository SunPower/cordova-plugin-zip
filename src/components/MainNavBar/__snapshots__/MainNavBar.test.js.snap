// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainNavBar component hides bell notification if there are no alerts 1`] = `
<nav
  aria-label="main navigation"
  className="main-nav-bar navbar"
  role="navigation"
>
  <div
    className="header-container columns is-mobile"
  >
    <div
      className="column"
    >
      <Link
        className="link"
        to={
          Object {
            "pathname": "/menu",
            "state": Object {
              "from": "/",
            },
          }
        }
      >
        <ForwardRef(SvgHamburgerMenu) />
      </Link>
    </div>
    <div
      className="column is-two-thirds header is-flex"
    >
      <h1
        className="navbar-header"
      >
        home
      </h1>
    </div>
    <div
      className="column actions"
    >
      <Link
        to={
          Object {
            "pathname": "/notifications",
            "state": Object {
              "from": "/",
            },
          }
        }
      >
        <div
          className="alerts-bell pr-5"
        >
          <ForwardRef(SvgAlerts) />
          <div
            className="is-hidden"
          />
        </div>
      </Link>
      <Link
        to={
          Object {
            "pathname": "/profile",
            "state": Object {
              "from": "/",
            },
          }
        }
      >
        <div
          className="pl-5"
        >
          <ForwardRef(SvgProfile) />
        </div>
      </Link>
    </div>
  </div>
</nav>
`;

exports[`MainNavBar component renders without crashing 1`] = `
<nav
  aria-label="main navigation"
  className="main-nav-bar navbar"
  role="navigation"
>
  <div
    className="header-container columns is-mobile"
  >
    <div
      className="column"
    >
      <Link
        className="link"
        to={
          Object {
            "pathname": "/menu",
            "state": Object {
              "from": "/",
            },
          }
        }
      >
        <ForwardRef(SvgHamburgerMenu) />
      </Link>
    </div>
    <div
      className="column is-two-thirds header is-flex"
    >
      <h1
        className="navbar-header"
      >
        true's home
      </h1>
    </div>
    <div
      className="column actions"
    >
      <Link
        to={
          Object {
            "pathname": "/notifications",
            "state": Object {
              "from": "/",
            },
          }
        }
      >
        <div
          className="alerts-bell pr-5"
        >
          <ForwardRef(SvgAlerts) />
          <div
            className="has-alerts"
          />
        </div>
      </Link>
      <Link
        to={
          Object {
            "pathname": "/profile",
            "state": Object {
              "from": "/",
            },
          }
        }
      >
        <div
          className="pl-5"
        >
          <ForwardRef(SvgProfile) />
        </div>
      </Link>
    </div>
  </div>
</nav>
`;
